File: dart_cli/lib/ioClass.dart

import "ByteDataWrapper.dart";
import "KeysLookup.dart";

abstract class IoClass {
  const IoClass();
  IoClass.readBytes(ByteDataWrapper bytes);
  void writeBytes(ByteDataWrapper bytes);

  int get size;
}
abstract class JsonIoClass {
  const JsonIoClass();
  JsonIoClass.fromJson(Object? json);
  JsonIoClass.readBytes(ByteDataWrapper bytes, KeysLookup keysLookup);
  
  Object? toJson();
  void writeBytes(ByteDataWrapper bytes, KeysLookup keysLookup);
  
  int get size;
}

File: python_lib/ioClass.py
from DataType import DataType
from FileWrapper import BinaryReader, BinaryWriter
from KeysLookup import KeysLookup


class JsonIoClass:
	@staticmethod
	def fromJson(json: object, keysLookup: KeysLookup):
		raise NotImplementedError()

	@staticmethod
	def readBytes(bytes: BinaryReader, dataType: DataType, keysLookup: KeysLookup):
		raise NotImplementedError()

	def toJson(self) -> object:
		raise NotImplementedError()

	def writeBytes(self, bytes: BinaryWriter, keysLookup: KeysLookup):
		raise NotImplementedError()

	@property
	def size(self) -> int:
		raise NotImplementedError()
